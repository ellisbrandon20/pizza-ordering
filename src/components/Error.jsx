import { Button, Div, Icon, Text } from "atomize";

export const Error = () => {

	return (
		<>
			<Div
				transform='translateY(10rem)'
				w="50rem"
				h="auto"
				m={{ l: 'auto', r: 'auto' }}
				p={{ t: '2rem', b: '2rem' }}
				rounded="lg"
				className="transparent-container"
			>
				<Text tag="h2" textSize="display1">Error: Please Try Again Later</Text>

				<Icon name="AlertSolid" size="5rem" color="danger800" m={{ t: '1rem' }} />
			</Div>
		</>
	);
};