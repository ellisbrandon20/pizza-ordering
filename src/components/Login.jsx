import { Button, Container, Icon, Input, Text } from "atomize";
import { Component } from "react";
import { getLoginToken } from "../apis";
import JWTMgr from "../JWTMgr";

export class Login extends Component {
	constructor(props) {
		super(props);
		this.state = {
			username: "",
			password: ""
		};
	}

	handleSubmit = (e) => {
		e.preventDefault();

		const { username, password } = this.state;
		getLoginToken(username, password)
			.then(responseData => {
				JWTMgr.setToken(responseData.access_token);
				this.props.goToScreen('Dashboard');
			}).catch(_ => {
				this.props.goToScreen('Error');
			});
	}

	handleChange = (input, e) => {
		this.setState({ [input]: e.target.value });
	}

	render() {
		return (
			<>
				<Container
					w="50rem"
					h="auto"
					m={{ l: 'auto', r: 'auto' }}
					p={{ t: '2rem', b: '2rem' }}
					transform='translateY(5rem)'
					rounded="lg"
					className="transparent-container"
				>
					<Text tag="h2" textSize="display1">Login</Text>
					<form onSubmit={this.handleSubmit}>
						<Input
							placeholder="Username"
							value={this.state.username}
							onChange={(e) => this.handleChange('username', e)}
							required
							w="18rem"
							p={{ x: "2.5rem" }}
							m={{ l: 'auto', r: 'auto', t: '0.5rem' }}
							d="block"
							borderColor="danger500"
							focusBorderColor="danger700"
							hoverBorderColor="danger700"
							prefix={
								<Icon
									name="UserSolid"
									color="danger700"
									cursor="pointer"
									pos="relative"
									transform="translate(140%, 15%)"
								/>
							}
						/>
						<Input
							placeholder="Password"
							type="password"
							value={this.state.password}
							onChange={(e) => this.handleChange('password', e)}
							required
							w="18rem"
							p={{ x: "2.5rem" }}
							m={{ l: 'auto', r: 'auto', t: '0.5rem' }}
							d="block"
							borderColor="danger500"
							focusBorderColor="danger700"
							hoverBorderColor="danger700"
							prefix={
								<Icon
									name="LockSolid"
									color="danger700"
									cursor="pointer"
									pos="relative"
									transform="translate(140%, 15%)"
								/>
							}
						/>
						<Button
							type="submit"
							rounded="circle"
							bg="danger700"
							hoverBg="danger600"
							m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
							shadow="2"
							hoverShadow="4"
						>
							Login
						</Button>
					</form>
				</Container>
			</>
		);
	}
}