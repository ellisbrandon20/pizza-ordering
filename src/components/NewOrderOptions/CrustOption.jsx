import { Div, Label, Radiobox, Text } from "atomize";

export const CrustOption = ({ crust, index, changeSelection }) => {
	return (
		<>
			<Text tag="p" textSize="subheader">Pick your Crust:</Text>
			<Div m={{ l: 'auto', r: 'auto', t: '0.5rem' }} w="18rem">
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Crust', index, 'Original')}
						checked={crust === 'Original'}
						name="crust"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Original
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Crust', index, 'Thin')}
						checked={crust === 'Thin'}
						name="crust"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Thin
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Crust', index, 'Stuffed')}
						checked={crust === 'Stuffed'}
						name="crust"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Stuffed
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Crust', index, 'Deepdish')}
						checked={crust === 'Deepdish'}
						name="crust"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Deepdish
				</Label>
			</Div>
		</>
	);
}