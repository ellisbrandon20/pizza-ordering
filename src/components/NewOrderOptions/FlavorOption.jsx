import { Div, Label, Radiobox, Text } from "atomize";


export const FlavorOption = ({ flavor, index, changeSelection }) => {
	return (
		<>
			<Text tag="p" textSize="subheader">Pick your Flavor:</Text>
			<Div m={{ l: 'auto', r: 'auto', t: '0.5rem' }} w="18rem">
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Flavor', index, 'Cheese')}
						checked={flavor === 'Cheese'}
						name="flavor"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Cheese
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Flavor', index, 'Pepperoni')}
						checked={flavor === 'Pepperoni'}
						name="flavor"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Pepperoni
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Flavor', index, 'Sausage')}
						checked={flavor === 'Sausage'}
						name="flavor"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Sausage
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Flavor', index, 'Supreme')}
						checked={flavor === 'Supreme'}
						name="flavor"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Supreme
				</Label>
			</Div>
		</>
	);
}