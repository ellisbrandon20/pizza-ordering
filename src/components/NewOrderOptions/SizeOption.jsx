import { Div, Label, Radiobox, Text } from "atomize";

export const SizeOption = ({ size, index, changeSelection }) => {
	return (
		<>
			<Text tag="p" textSize="subheader">Pick your Size:</Text>
			<Div m={{ l: 'auto', r: 'auto', t: '0.5rem' }} w="18rem">
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Size', index, 'S')}
						checked={size === 'S'}
						name="size"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Small (12")
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Size', index, 'M')}
						checked={size === 'M'}
						name="size"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Medium (16")
				</Label>
				<Label align="center" textWeight="600" m={{ b: "0.5rem", r: "2rem" }}>
					<Radiobox
						onChange={(e) => changeSelection(e, 'Size', index, 'L')}
						checked={size === 'L'}
						name="size"
						inactiveColor="danger500"
						activeColor="danger700"
					/>
					Large (18")
				</Label>
			</Div>
		</>
	);
}