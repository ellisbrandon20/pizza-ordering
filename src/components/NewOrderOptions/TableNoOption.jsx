import { Div, Input, Text } from "atomize";


export const TabelNoOption = ({ tableNo, index, changeSelection }) => {
	return (
		<>
			<Text tag="p" textSize="subheader">Table Number:</Text>
			<Div m={{ l: 'auto', r: 'auto', t: '0.5rem' }} w="18rem">
				<Input
					placeholder="Table #"
					name='tableNo'
					type='number'
					onChange={(e) => changeSelection(e, 'Table_No', index)}
					w='5.75rem'
					borderColor="danger500"
					focusBorderColor="danger700"
				/>
			</Div>
		</>
	);
}