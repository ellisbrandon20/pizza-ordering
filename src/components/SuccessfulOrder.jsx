import { Button, Div, Icon, Text } from "atomize";

export const SuccessfulOrder = ({ goToScreen }) => {

	return (
		<>
			<Div
				transform='translateY(10rem)'
				w="50rem"
				h="auto"
				m={{ l: 'auto', r: 'auto' }}
				p={{ t: '2rem', b: '2rem' }}
				rounded="lg"
				className="transparent-container"
			>
				<Text tag="h2" textSize="display1">Successfully Submitted Order</Text>

				<Icon name="Checked" size="5rem" color="black" />
				<Button
					onClick={() => goToScreen('Dashboard')}
					prefix={
						<Icon
							name="LongLeft"
							size="16px"
							color="white"
							m={{ r: "0.5rem" }}
						/>
					}
					bg="danger700"
					hoverBg="danger600"
					rounded="circle"
					p={{ r: "1.5rem", l: "1rem" }}
					m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
					shadow="2"
					hoverShadow="4"
				>
					Back To Dashboard
				</Button>
			</Div>
		</>
	);
};