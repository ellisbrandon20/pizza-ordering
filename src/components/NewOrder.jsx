import { Button, Col, Container, Div, Icon, Row, Text } from "atomize";
import { Component } from "react";
import { createOrder } from "../apis";
import { CrustOption } from "./NewOrderOptions/CrustOption";
import { FlavorOption } from "./NewOrderOptions/FlavorOption";
import { SizeOption } from "./NewOrderOptions/SizeOption";
import { TabelNoOption } from "./NewOrderOptions/TableNoOption";

export class NewOrder extends Component {
	constructor(props) {
		super(props);

		this.state = {
			'newOrders': [{
				'Crust': '',
				'Flavor': '',
				'Size': '',
				'Table_No': undefined,
			}]
		};
	}

	changeSelection = (e, option, index, value) => {
		if (option === 'Table_No') value = parseInt(e.target.value);

		let pizzaOrders = this.state.newOrders;

		pizzaOrders[index][option] = value;

		this.setNewOrders(pizzaOrders);
	};

	handleSubmit = (e) => {
		e.preventDefault();
		const { newOrders } = this.state;

		let successOrder = false;
		newOrders.forEach(order => {
			successOrder = this.createPizzaOrder(order);
		});

		if (successOrder) this.props.goToScreen("SuccessfulOrder");
	}

	createPizzaOrder = (order) => {
		return createOrder(order)
			.then(_ => {
				return true;
			}).catch(_ => {
				this.props.goToScreen('Error');
				return false;
			});
	}

	addPizzaOrder = (e) => {
		e.preventDefault();

		let pizzaOrders = this.state.newOrders;
		pizzaOrders.push({
			'Crust': '',
			'Flavor': '',
			'Size': '',
			'Table_No': undefined,
		});

		this.setNewOrders(pizzaOrders);
	}

	deleteNewOrder = (e, index) => {
		e.preventDefault();

		let pizzaOrders = this.state.newOrders;
		pizzaOrders.splice(index, 1);

		this.setNewOrders(pizzaOrders);
	}

	setNewOrders = (pizzaOrders) => {
		this.setState({
			'newOrders': pizzaOrders
		});
	}

	render() {
		const { newOrders } = this.state;
		return (
			<Div transform='translateY(5rem)'>
				<Text tag="h2" textSize="display1">New Pizza Order</Text>

				<form id='newOrder' onSubmit={this.handleSubmit}>
					{newOrders.map((order, i) => {
						return (
							<Container
								key={i}
								w="50rem"
								h="28rem"
								m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
								rounded="lg"
								className="transparent-container"
							>
								<Text key={`OrderNo-${i}`} tag="h3" textSize="heading">Pizza #{i + 1}</Text>

								<Row>
									<Col size='6'>
										<CrustOption
											key={`crust-${i}`}
											crust={order.Crust}
											index={i}
											changeSelection={this.changeSelection}
										/>
									</Col>
									<Col size='6'>
										<FlavorOption
											key={`flavor-${i}`}
											flavor={order.Flavor}
											index={i}
											changeSelection={this.changeSelection}
										/>
									</Col>
								</Row>
								<Row>
									<Col size='6'>
										<SizeOption
											key={`size-${i}`}
											size={order.Size}
											index={i}
											changeSelection={this.changeSelection}
										/>

									</Col>
									<Col size='6'>
										<TabelNoOption
											key={`tableNo-${i}`}
											tableNo={order.Table_No}
											index={i}
											changeSelection={this.changeSelection}
										/>

									</Col>
								</Row>
								<Row>
									<Col size='12'>
										<Div d="flex">
											<Button
												onClick={(e) => this.addPizzaOrder(e)}
												h="2.5rem"
												w="2.5rem"
												bg="danger700"
												hoverBg="danger600"
												rounded="circle"
												m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
												shadow="2"
												hoverShadow="4"
											>
												<Icon name="Add" size="20px" color="white" />
											</Button>
											<Button
												onClick={(e) => this.deleteNewOrder(e, i)}
												h="2.5rem"
												w="2.5rem"
												bg="danger700"
												hoverBg="danger600"
												rounded="circle"
												m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
												shadow="2"
												hoverShadow="4"
											>
												<Icon name="DeleteSolid" size="20px" color="white" />
											</Button>
										</Div>
									</Col>
								</Row>
							</Container>
						);
					})}

					<Button
						type='submit'
						h="2.5rem"
						w="8rem"
						bg="success800"
						hoverBg="success700"
						rounded="circle"
						m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
						shadow="2"
						hoverShadow="4"
					>
						Place Order
					</Button>
				</form>
			</Div>
		);
	}
}