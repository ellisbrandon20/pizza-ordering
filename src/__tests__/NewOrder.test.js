import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Container, Text } from 'atomize';
import { NewOrder } from '../components/NewOrder';
import { CrustOption } from '../components/NewOrderOptions/CrustOption';
import { FlavorOption } from '../components/NewOrderOptions/FlavorOption';
import { SizeOption } from '../components/NewOrderOptions/SizeOption';
import { TabelNoOption } from '../components/NewOrderOptions/TableNoOption';
Enzyme.configure({ adapter: new Adapter() });

test('NewOrder Snapshot', () => {
	const wrapper = renderer.create(<NewOrder />).toJSON();
	expect(wrapper).toMatchSnapshot();
});

test('should render form', () => {
	const component = shallow(<NewOrder />);
	const form = component.find('#newOrder');
	expect(form).toHaveLength(1);
});

test('should render single new Pizza Order Container and Fields', () => {
	const component = shallow(<NewOrder />);

	const container = component.find(Container);
	expect(container).toHaveLength(1);

	const newOrderHeader = component.find(Text);
	expect(newOrderHeader).toHaveLength(2);
	expect(newOrderHeader.at(1).text()).toEqual('Pizza #1');

	const crustField = component.find(CrustOption);
	expect(crustField).toHaveLength(1);

	const flavorOption = component.find(FlavorOption);
	expect(flavorOption).toHaveLength(1);

	const sizeOption = component.find(SizeOption);
	expect(sizeOption).toHaveLength(1);

	const tableNoOption = component.find(TabelNoOption);
	expect(tableNoOption).toHaveLength(1);
});

test('should render multiple new Pizza Order Container and Fields', () => {
	const component = shallow(<NewOrder />);
	component.setState({
		'newOrders': [
			{
				'Crust': '',
				'Flavor': '',
				'Size': '',
				'Table_No': undefined,
			},
			{
				'Crust': '',
				'Flavor': '',
				'Size': '',
				'Table_No': undefined,
			}
		]
	});

	const container = component.find(Container);
	expect(container).toHaveLength(2);

	const newOrderHeader = component.find(Text);
	expect(newOrderHeader).toHaveLength(3);
	expect(newOrderHeader.at(1).text()).toEqual('Pizza #1');
	expect(newOrderHeader.at(2).text()).toEqual('Pizza #2');

	const crustField = component.find(CrustOption);
	expect(crustField).toHaveLength(2);

	const flavorOption = component.find(FlavorOption);
	expect(flavorOption).toHaveLength(2);

	const sizeOption = component.find(SizeOption);
	expect(sizeOption).toHaveLength(2);

	const tableNoOption = component.find(TabelNoOption);
	expect(tableNoOption).toHaveLength(2);
});