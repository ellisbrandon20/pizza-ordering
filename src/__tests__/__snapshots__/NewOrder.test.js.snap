// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewOrder Snapshot 1`] = `
<div
  className=""
>
  <h2
    className=""
  >
    New Pizza Order
  </h2>
  <form
    id="newOrder"
    onSubmit={[Function]}
  >
    <div
      className="transparent-container "
    >
      <h3
        className=""
      >
        Pizza #
        1
      </h3>
      <div
        className=""
      >
        <div
          className=""
        >
          <p
            className=""
          >
            Pick your Crust:
          </p>
          <div
            className=""
          >
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="crust"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Original
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="crust"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Thin
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="crust"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Stuffed
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="crust"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Deepdish
            </label>
          </div>
        </div>
        <div
          className=""
        >
          <p
            className=""
          >
            Pick your Flavor:
          </p>
          <div
            className=""
          >
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="flavor"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Cheese
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="flavor"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Pepperoni
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="flavor"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Sausage
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="flavor"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Supreme
            </label>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        >
          <p
            className=""
          >
            Pick your Size:
          </p>
          <div
            className=""
          >
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="size"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Small (12")
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="size"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Medium (16")
            </label>
            <label
              className=""
              tabIndex="0"
            >
              <input
                checked={false}
                className=""
                cursor="pointer"
                name="size"
                onChange={[Function]}
                type="radio"
              />
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FABEAA"
                />
              </svg>
              Large (18")
            </label>
          </div>
        </div>
        <div
          className=""
        >
          <p
            className=""
          >
            Table Number:
          </p>
          <div
            className=""
          >
            <div
              className=""
            >
              <input
                className=""
                d="flex"
                name="tableNo"
                onChange={[Function]}
                placeholder="Table #"
                type="number"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        >
          <div
            className=""
          >
            <button
              className=""
              onClick={[Function]}
            >
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,11 L8,11 C7.44771525,11 7,11.4477153 7,12 C7,12.5522847 7.44771525,13 8,13 L11,13 L11,16 C11,16.5522847 11.4477153,17 12,17 C12.5522847,17 13,16.5522847 13,16 L13,13 L16,13 C16.5522847,13 17,12.5522847 17,12 C17,11.4477153 16.5522847,11 16,11 L13,11 L13,8 C13,7.44771525 12.5522847,7 12,7 C11.4477153,7 11,7.44771525 11,8 L11,11 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
                  fill="#FFFFFF"
                  fillRule="evenodd"
                />
              </svg>
            </button>
            <button
              className=""
              onClick={[Function]}
            >
              <svg
                className=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M5,7 L4,7 C3.44771525,7 3,6.55228475 3,6 C3,5.44771525 3.44771525,5 4,5 L8,5 L8,3 C8,2.44771525 8.44771525,2 9,2 L15,2 C15.5522847,2 16,2.44771525 16,3 L16,5 L20,5 C20.5522847,5 21,5.44771525 21,6 C21,6.55228475 20.5522847,7 20,7 L19,7 L19,20.5 C19,21.3284271 18.3284271,22 17.5,22 L6.5,22 C5.67157288,22 5,21.3284271 5,20.5 L5,7 Z M11,17 L11,11 C11,10.4477153 10.5522847,10 10,10 C9.44771525,10 9,10.4477153 9,11 L9,17 C9,17.5522847 9.44771525,18 10,18 C10.5522847,18 11,17.5522847 11,17 Z M15,17 L15,11 C15,10.4477153 14.5522847,10 14,10 C13.4477153,10 13,10.4477153 13,11 L13,17 C13,17.5522847 13.4477153,18 14,18 C14.5522847,18 15,17.5522847 15,17 Z"
                  fill="#FFFFFF"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <button
      className=""
      type="submit"
    >
      Place Order
    </button>
  </form>
</div>
`;
